File: frontend/youthconnect/src/app/services/youth-profile.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { YouthProfile, YouthProfileWithClassification, YouthProfileDTO } from '../models/youth-profile.model';

@Injectable({
    providedIn: 'root'
})
export class YouthProfileService {
    private apiUrl = 'http://localhost:8080/api/youth-profile';

    constructor(private http: HttpClient) { }

    registerYouth(profile: YouthProfile): Observable<YouthProfile> {
        return this.http.post<YouthProfile>(`${this.apiUrl}/register`, profile);
    }

    getAllYouthProfiles(): Observable<YouthProfile[]> {
        return this.http.get<YouthProfile[]>(this.apiUrl);
    }
    
    getAllYouthProfilesWithClassification(): Observable<YouthProfileWithClassification[]> {
        return this.http.get<YouthProfileWithClassification[]>(`${this.apiUrl}/with-classification`);
    }

    getYouthProfileById(id: number): Observable<YouthProfile> {
        return this.http.get<YouthProfile>(`${this.apiUrl}/${id}`);
    }
    
    getYouthProfileWithClassificationById(id: number): Observable<YouthProfileWithClassification> {
        return this.http.get<YouthProfileWithClassification>(`${this.apiUrl}/${id}/with-classification`);
    }

    updateYouthProfile(id: number, profile: YouthProfile): Observable<YouthProfile> {
        return this.http.put<YouthProfile>(`${this.apiUrl}/${id}`, profile);
    }
    
    updateYouthProfileWithClassification(id: number, profile: YouthProfileDTO): Observable<YouthProfile> {
        return this.http.put<YouthProfile>(`${this.apiUrl}/${id}/with-classification`, profile);
    }

    deleteYouthProfile(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}`);
    }
}

File: backend/youthconnect_id/src/main/java/com/youthconnect/youthconnect_id/services/YouthProfileService.java
package com.youthconnect.youthconnect_id.services;

import com.youthconnect.youthconnect_id.models.YouthProfile;
import com.youthconnect.youthconnect_id.models.dto.YouthProfileDTO;
import java.util.List;
import java.util.Optional;

public interface YouthProfileService {
    YouthProfile registerYouth(YouthProfileDTO youthProfileDTO);
    
    List<YouthProfile> getAllYouthProfiles();
    
    Optional<YouthProfile> getYouthProfileById(Integer id);
    
    YouthProfile updateYouthProfile(Integer id, YouthProfileDTO youthProfileDTO);
    
    void deleteYouthProfile(Integer id);
}
